### DISCUSSION TITLE
Widget Entry with a custom icon

### DISCUSSION THREAD

**Discussion by aneeskA** [fyne-io/fyne#4924] [Q&A] [answered]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same feature, before opening a new one.
- [X] This issue only relates to a single feature. I will open new issues for any other features.

### Is your feature request related to a problem?

Widget Entry allows `Password` to be set to `true/false` which adds an interactive icon at the right side of the widget. Is it possible to extend this so that we can pass in any icon and a listener to it(optional)? This way I can create a search bar with a search icon in the Entry widget itself.

### Describe the solution you'd like to see.

NA

**Comment by aneeskA** [comment, marked as answer]
Argh! I discovered it a few minutes after posting the question.

```
widget.Entry{PlaceHolder: "Search", ActionItem: widget.NewIcon(theme.SearchIcon()),}
```

**Comment by aneeskA** [comment]
Apologies.

**Comment by Jacalz** [comment]
I’m glad that you found the solution. However, please do use GitHub Discussions for questions instead next time. I’ll convert this to an discussion :)
